### PR TITLE
7148092: [macosx] When Alt+down arrow key is pressed, the combobox popup does not appear.

### PR DESCRIPTION
**Pull request by prsadhuk**
When comboBox have focus, pressing Alt+Down (or Up) should render the popup which is not happening for Aqua L&F.

Fix is to add toggleAction for Alt+Down/Up keys as has been done for [MetalLookAndFeel](https://github.com/openjdk/jdk/blob/master/src/java.desktop/share/classes/javax/swing/plaf/metal/MetalLookAndFeel.java#L912)

closed problemlisted test is used to check the fix.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-7148092](https://bugs.openjdk.org/browse/JDK-7148092): [macosx] When Alt+down arrow key is pressed, the combobox popup does not appear.

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [531dec1f](https://git.openjdk.org/jdk/pull/10014/files/531dec1f8db3fd506bae37072dc523a761eee5d5)
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**) ⚠️ Review applies to [925c37d1](https://git.openjdk.org/jdk/pull/10014/files/925c37d1ac5ad9ea419abc4bf8964a2c4b5d05d2)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10014/head:pull/10014` \
`$ git checkout pull/10014`

Update a local copy of the PR: \
`$ git checkout pull/10014` \
`$ git pull https://git.openjdk.org/jdk pull/10014/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10014`

View PR using the GUI difftool: \
`$ git pr show -t 10014`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10014.diff">https://git.openjdk.org/jdk/pull/10014.diff</a>

</details>

### CODE REVIEW

**Comment by prsadhuk** [comment]
It seems Aua native combox does something extra. 
When Alt+Down is pressed on combobox, it opens up the popup but if the popup is already opened, it selects the last entry.
When Alt+Up is pressed on combobox, it opens up the popup but if the popup is already opened, it selects the first entry.

Modified the fix to behave same as native combobox.

File: src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxUI.java [hunk 1]
```diff
@@ -581,6 +583,35 @@ public boolean isEnabled() {
         }
     };
 
+    @SuppressWarnings("serial") // anonymous class
+    private final Action openPopupOrhighlightLast = new AbstractAction() {
```
Comment (andy-goryachev-oracle) [inline src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxUI.java:587]: may be openPopupOrHighlightLast (uppercase H)?

File: src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxUI.java [hunk 2]
```diff
@@ -581,6 +583,35 @@ public boolean isEnabled() {
...
+            } else {
+                final int size = listBox.getModel().getSize();
+                listBox.setSelectedIndex(size - 1);
+                listBox.ensureIndexIsVisible(size - 1);
+            }
+        }
+    };
+
+    @SuppressWarnings("serial") // anonymous class
+    private final Action openPopupOrhighlightFirst = new AbstractAction() {
```
Comment (andy-goryachev-oracle) [inline src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxUI.java:602]: may be openPopupOrHighlightFirst (uppercase H)

Comment (prsadhuk) [reply src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxUI.java]: ok

**Review by prrace — APPROVED** [review]
Looks OK to me ..  I presume the closed test now passes . .

**Comment by prsadhuk** [comment]
> Looks OK to me .. I presume the closed test now passes . .

Yes, it passes. Will you also review the closed part in orahub as it involves removing the test from PL?

**Comment by mrserb** [comment]
Don't we need to align this behaviour to the native apps rather than metal?

**Comment by prsadhuk** [comment]
Native behaves same as is being done in this PR (which is extra from Metal) which is already mentioned above.

**Comment by mrserb** [comment]
Then the test mentioned above should be updated as well to cover that extra?

**Review by mrserb — APPROVED** [review]

**Pull request closed without merging**